### PR TITLE
Enabled server-side secure grpc for connections established with VM

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian <eswar@accuknox.com>

### CODE REVIEW

**Review by nyrahul — APPROVED** [review]
LGTM

File: KubeArmor/kvmAgent/kvmAgent.go [hunk 1]
```diff
@@ -53,9 +59,22 @@ func NewKVMAgent(eventCb tp.KubeArmorHostPolicyEventCallback) *KVMAgent {
...
+		InsecureSkipVerify: false,
+		RootCAs:            cp,
+		ServerName:         "0.0.0.0",
+		MinVersion:         tls.VersionTLS12,
+	}
+
 	// Connect to gRPC server
 	ctx, _ := context.WithTimeout(context.Background(), time.Second*5)
-	gRPCConnection, err := grpc.DialContext(ctx, kvm.gRPCServer, grpc.WithInsecure(), grpc.WithBlock())
+	gRPCConnection, err := grpc.DialContext(ctx, kvm.gRPCServer, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)), grpc.WithBlock())
```
Comment (nam-jaehyun) [inline KubeArmor/kvmAgent/kvmAgent.go:77]: Can we add the case if certs do not exist as exception handling?
In this implementation, it seems that KubeArmor will fail if no certs.
I feel when people test KubeArmor locally, they may not have any certs.

**Review by nam-jaehyun — CHANGES_REQUESTED** [review]
we may need to handle possible corner cases.

File: KubeArmor/kvmAgent/kvmAgent.go (hunk 1 above)
Comment (seswarrajan) [reply KubeArmor/kvmAgent/kvmAgent.go:77]: Yes @nam-jaehyun. It is possible. Is it like when certs are not available, we have to establish connection in insecure mode ?
We can use InSecureSkipVerify in TLSConfig if certs are not available or we can use grpc.WithInSecure() option in Dial.

**Review by nam-jaehyun — COMMENTED** [review]
was this PR merged into the main branch?

**Comment by seswarrajan** [comment]
Currently closing the same as this has issue loading certificate. Will handle later.
